### PR TITLE
feat(permissions): Delete permissions: (DSP-1169)

### PR DESCRIPTION
**Pull request by SepidehAlassi**
resolves DSP-1169
Additionally:
- [x] For all permission operations systematically check requesting user's permission for the operation + tests

### CODE REVIEW

File: webapi/src/main/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/PermissionsMessagesADM.scala [hunk 1]
```diff
@@ -697,6 +703,30 @@ case class PermissionByIriGetRequestADM(permissionIri: IRI, requestingUser: User
...
+  *
+  * @param permissionIri               the iri of the permission object.
+  * @param requestingUser              the user initiating the request.
+  * @param apiRequestID                the API request ID.
+  */
+case class PermissionDeleteRequestADM(permissionIri: IRI, requestingUser: UserADM, apiRequestID: UUID)
+    extends PermissionsResponderRequestADM {
+  // Check user's permission for the operation
+  if (!requestingUser.isSystemAdmin
+      && !requestingUser.permissions.isProjectAdminInAnyProject()
```
Comment (subotic) [inline webapi/src/main/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/PermissionsMessagesADM.scala:719]: Isn't it necessary for the `requestingUser` to be `ProjectAdmin` of the project that the permission belongs to?

File: webapi/src/test/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/PermissionsMessagesADMSpec.scala [hunk 2]
```diff
@@ -57,14 +57,15 @@ class PermissionsMessagesADMSpec extends CoreSpec() {
...
+          administrativePermissionIri = permissionIri,
           requestingUser = SharedTestDataADM.imagesUser01,
           apiRequestID = UUID.randomUUID()
         )
       )
-      assert(caught.getMessage === "Invalid permission IRI")
+      assert(caught.getMessage === s"Invalid permission IRI $permissionIri is given.")
     }
 
     "return 'ForbiddenException' if the user requesting AdministrativePermissionForIriGetRequestADM is not SystemAdmin" in {
```
Comment (subotic) [inline webapi/src/test/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/PermissionsMessagesADMSpec.scala:71]: Test for not ProjectAdmin is missing.

**Review by subotic — CHANGES_REQUESTED** [review]

File: webapi/src/test/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/PermissionsMessagesADMSpec.scala (hunk 2 above)
Comment (SepidehAlassi) [reply webapi/src/test/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/PermissionsMessagesADMSpec.scala]: the title of the test is wrong, it checks for the project admin.

File: webapi/src/main/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/PermissionsMessagesADM.scala (hunk 1 above)
Comment (SepidehAlassi) [reply webapi/src/main/scala/org/knora/webapi/messages/admin/responder/permissionsmessages/PermissionsMessagesADM.scala]: Thanks for pointing this out, I limited this to project admin.

**Review by subotic — APPROVED** [review]
Awesome, thanks :-)

**Comment by SepidehAlassi** [comment]
@subotic Thanks for reviewing this!

**Pull request merged**